app/services/sync_provider_from_find.rb

```diff
@@ -78,12 +82,13 @@ def self.create_or_update_course(find_course, provider)
         end
 
       study_modes.each do |mode|
-        CourseOption.find_or_create_by(
+        course_option = CourseOption.find_or_create_by(
           site: site,
           course_id: course.id,
           study_mode: mode,
-          vacancy_status: 'B', #TODO: Should this be reflected by `find_course.has_vacancies?`
         )
+
+        course_option.update(vacancy_status: site_status.vac_status)
```

Comment:
```suggestion
        course_option.update!(vacancy_status: site_status.vac_status)
```

To avoid silent failures (if we introduce validations or something).